cmd/iam.go

```diff
@@ -1664,15 +1681,32 @@ func (sys *IAMSys) IsAllowedSTS(args iampolicy.Args) bool {
 		// No policy available reject.
 		return false
 	}
-	name := mp.Policy
 
-	if pnameSlice[0] != name {
+	if strings.Join(policies, ",") != mp.Policy {
 		// When claims has a policy, it should match the
 		// policy of args.AccountName which server remembers.
 		// if not reject such requests.
 		return false
 	}
 
+	var availablePolicies []iampolicy.Policy
+	for _, pname := range policies {
+		p, found := sys.iamPolicyDocsMap[pname]
+		if found {
```

Comment:
I guess it is okay to do not throw an error if a policy is not found, since only the admin can delete policies, right ?

Reply:
> I guess it is okay to do not throw an error if a policy is not found, since only the admin can delete policies, right ?

Maybe this needs to return an error instead. No harmful vs more security.

Reply:
correct @vadmeste